Source/Core/Core/WC24PatchEngine.cpp

```diff
@@ -0,0 +1,159 @@
+    Titles::NINTENDO_CHANNEL_NTSC_J,
+    Titles::NINTENDO_CHANNEL_PAL,
+    Titles::FORECAST_CHANNEL_NTSC_U,
+    Titles::FORECAST_CHANNEL_NTSC_J,
+    Titles::FORECAST_CHANNEL_PAL,
+    Titles::NEWS_CHANNEL_NTSC_U,
+    Titles::NEWS_CHANNEL_NTSC_J,
+    Titles::NEWS_CHANNEL_PAL,
+    Titles::EVERYBODY_VOTES_CHANNEL_NTSC_U,
+    Titles::EVERYBODY_VOTES_CHANNEL_NTSC_J,
+    Titles::EVERYBODY_VOTES_CHANNEL_PAL,
+};
+
+static std::vector<NetworkPatch> s_patches;
+
+bool DeserializeLine(const std::string& line, NetworkPatch* patch)
+{
+  const std::vector<std::string> items = SplitString(line, ':');
+
+  if (items.size() != 3)
+    return false;
+
+  patch->source = items[0];
+  patch->replacement = items[1];
+
+  if (!TryParse(items[2], &patch->is_kd))
+    return false;
+
+  return patch;
+}
+
+void LoadPatchSection(const Common::IniFile& ini)
+{
+  std::vector<std::string> lines;
+  NetworkPatch patch;
+  ini.GetLines("WC24Patch", &lines);
+
+  for (std::string& line : lines)
+  {
+    if (line.empty())
+      continue;
+
+    if (line[0] == '$')
+    {
+      patch.name = line.substr(1, line.size() - 1);
+    }
+    else
+    {
+      if (DeserializeLine(line, &patch))
+      {
+        s_patches.push_back(patch);
+      }
+    }
+  }
+
+  ReadEnabledAndDisabled(ini, "WC24Patch", &s_patches);
+}
+
+void LoadPatches()
+{
+  const auto& sconfig = SConfig::GetInstance();
+  // We can only load WC24 Channels.
+  if (!IsWC24Channel())
+    return;
+
+  Common::IniFile ini;
+  // If WiiLink is enabled then we load the Default Ini as that has the WiiLink URLs.
+  if (Config::Get(Config::MAIN_WII_WIILINK_ENABLE))
+    ini = sconfig.LoadDefaultGameIni();
+  else
+    ini = sconfig.LoadLocalGameIni();
+
+  LoadPatchSection(ini);
+}
+
+bool IsWC24Channel()
+{
+  const auto& sconfig = SConfig::GetInstance();
+  const auto found =
+      std::find(s_wc24_channels.begin(), s_wc24_channels.end(), sconfig.GetTitleID());
+
+  return found != s_wc24_channels.end();
+}
+
+void Reload()
+{
+  s_patches.clear();
+  LoadPatches();
+}
+
+std::optional<std::string> GetNetworkPatch(const std::string& source, IsKD is_kd)
+{
+  const auto patch =
+      std::find_if(s_patches.begin(), s_patches.end(), [&source, &is_kd](NetworkPatch& patch) {
+        return patch.source == source && patch.is_kd == is_kd;
+      });
+  if (patch == s_patches.end())
+    return std::nullopt;
+
+  return patch->replacement;
+}
+
+// When we patch for the Socket, we aren't given the URL. Instead, it is in a Host header
+// in the payload that the socket is going to send. We need to manually find which patch to apply.
+std::optional<std::string> GetNetworkPatchByPayload(std::string_view source)
+{
+  size_t pos{};
+  while (pos < source.size())
+  {
+    if (source.substr(pos).starts_with("Host:"))
+    {
+      size_t end_of_line = source.find("\r\n", pos);
+      std::string_view domain =
+          source.substr(pos + 6, end_of_line == std::string_view::npos ? std::string_view::npos :
```

Comment:
The `size_t end_of_line = source.find("\r\n", pos);` can be pulled out of the branch.

There's a mismatch here between the `starts_with("Host:")` and `.substr(pos + 6`, you're dropping a byte here. I'm not sure whether HTTP forces a space between the `:` and the start of the data? Might be okay to just change to `starts_with("Host: ")`?

Adding a few `const`s to variables here wouldn't hurt either.